lib/rialto/etl/cli/grants.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+require 'parallel'
+
+module Rialto
+  module Etl
+    module CLI
+      # Grants iterates over the people in the input file JSON and for each
+      # valid UID (SUNet ID) calls three other commands:
+      #   exe/extract call Sera --sunetid username > username.json
+      #   TODO: exe/transform call Sera -i username.json > username.sparql
+      #   TODO: exe/load call Sparql -i username.sparql
+      class Grants < Thor
+        option :input_file,
+               required: true,
+               banner: 'FILENAME',
+               desc: 'Name of file with data to be loaded (REQUIRED)',
+               aliases: '-i'
+        option :dir,
+               required: false,
+               default: 'data',
+               banner: 'DIR',
+               desc: 'Name of the directory to store data in',
+               aliases: '-d'
+        option :batch_size,
+               required: false,
+               default: 3,
+               banner: 'BATCH_SIZE',
+               desc: 'Size of batch for parallel processing',
+               aliases: '-s'
+        option :force,
+               required: false,
+               default: false,
+               banner: 'FORCE',
+               desc: 'Overwrite files that already exist',
+               aliases: '-f'
+
+        desc 'Load', 'Load all grants for all researchers in the file'
+        # rubocop:disable Metrics/MethodLength
+        # rubocop:disable Metrics/AbcSize
+        def load
+          FileUtils.mkdir_p(options[:dir])
+          File.open(options[:input_file], 'r') do |f|
+            f.each_line.lazy.each_slice(options[:batch_size].to_i) do |lines|
```

Comment:
🕶

Reply:
Found this on the Googles when searching for a pattern to stream *n* lines from a file at a time rather than have to hold a large file in memory all at once. Do you think it needs a comment or was the intent clear?